src/linear_algebra/orientation.lean

```diff
@@ -0,0 +1,379 @@
+   by rw [mul_smul, mul_smul, h₁, ←h₂, smul_comm]⟩
+
+/-- `same_ray` is reflexive. -/
+lemma reflexive_same_ray [nontrivial R] :
+  reflexive (same_ray R : M → M → Prop) :=
+λ _, ⟨1, 1, zero_lt_one, zero_lt_one, rfl⟩
+
+/-- `same_ray` is an equivalence relation. -/
+lemma equivalence_same_ray [nontrivial R] :
+  equivalence (same_ray R : M → M → Prop) :=
+⟨reflexive_same_ray R M, symmetric_same_ray R M, transitive_same_ray R M⟩
+
+variables {R M}
+
+/-- A vector is in the same ray as a positive multiple of itself. -/
+lemma same_ray_pos_smul_right (v : M) {r : R} (h : 0 < r) : same_ray R v (r • v) :=
+⟨r, 1, h, let f := nontrivial_of_lt _ _ h in by exactI zero_lt_one, (one_smul _ _).symm⟩
+
+/-- A vector is in the same ray as a positive multiple of one it is in the same ray as. -/
+lemma same_ray.pos_smul_right {v₁ v₂ : M} {r : R} (h : same_ray R v₁ v₂) (hr : 0 < r) :
+  same_ray R v₁ (r • v₂) :=
+transitive_same_ray R M h (same_ray_pos_smul_right v₂ hr)
+
+/-- A positive multiple of a vector is in the same ray as that vector. -/
+lemma same_ray_pos_smul_left (v : M) {r : R} (h : 0 < r) : same_ray R (r • v) v :=
+⟨1, r, let f := nontrivial_of_lt _ _ h in by exactI zero_lt_one, h, one_smul _ _⟩
+
+/-- A positive multiple of a vector is in the same ray as one it is in the same ray as. -/
+lemma same_ray.pos_smul_left {v₁ v₂ : M} {r : R} (h : same_ray R v₁ v₂) (hr : 0 < r) :
+  same_ray R (r • v₁) v₂ :=
+transitive_same_ray R M (same_ray_pos_smul_left v₁ hr) h
+
+variables (R M)
+
+/-- The setoid of the `same_ray` relation for elements of a module. -/
+def same_ray_setoid [nontrivial R] : setoid M :=
+{ r := λ v₁ v₂, same_ray R v₁ v₂, iseqv := equivalence_same_ray R M }
+
+/-- Nonzero vectors, as used to define rays. -/
+@[reducible] def ray_vector := {v : M // v ≠ 0}
+
+/-- The setoid of the `same_ray` relation for the subtype of nonzero vectors. -/
+def ray_vector.same_ray_setoid [nontrivial R] : setoid (ray_vector M) :=
+(same_ray_setoid R M).comap coe
+
+local attribute [instance] ray_vector.same_ray_setoid
+
+variables {R M}
+
+/-- Equivalence of nonzero vectors, in terms of same_ray. -/
+lemma equiv_iff_same_ray [nontrivial R] (v₁ v₂ : ray_vector M) :
+  v₁ ≈ v₂ ↔ same_ray R (v₁ : M) v₂ :=
+iff.rfl
+
+variables (R M)
+
+/-- A ray (equivalence class of nonzero vectors with common positive multiples) in a module. -/
+@[nolint has_inhabited_instance]
+def module.ray [nontrivial R] := quotient (ray_vector.same_ray_setoid R M)
+
+/-- An orientation of a module, intended to be used when `ι` is a `fintype` with the same
+cardinality as a basis. -/
+abbreviation orientation [nontrivial R] := module.ray R (alternating_map R M R ι)
+
+/-- A type class fixing an orientation of a module. -/
+class module.oriented [nontrivial R] :=
+(positive_orientation : orientation R M ι)
+
+variables {M}
+
+/-- The ray given by a nonzero vector. -/
+protected def ray_of_ne_zero [nontrivial R] {v : M} (h : v ≠ 0) : module.ray R M :=
+⟦⟨v, h⟩⟧
```

Comment:
I would recommend adding
```lean
/-- An induction rule for `module.ray`, used as `induction x using module.ray.ind`. -/
lemma module.ray.ind [nontrivial R] {C : module.ray R M → Sort*}
  (h : Π v (hv : v ≠ 0), C (ray_of_ne_zero R hv)) (x : module.ray R M) : C x :=
quotient.ind (subtype.rec $ by exact h) x
```
so that you can write `induction x using module.ray.ind` below.

Reply:
If I try this (adjusted for the newly explicit argument to `ray_of_ne_zero`), I get `"eliminator" elaborator failed to compute the motive`.

Reply:
Works fine for me in gitpod.